### PR TITLE
Changes is HLS master playlist

### PR DESCRIPTION
**Pull request by mcori**

### CODE REVIEW

**Comment by kaltura-hooks** [comment]
Hi @mcori,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

**Comment by kaltura-hooks** [comment]
Hi @mcori,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

**Comment by kaltura-hooks** [comment]
Hi @mcori,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: vod/hls/m3u8_builder.c [hunk 1]
```diff
@@ -511,48 +511,56 @@ m3u8_builder_build_master_playlist(
...
+			if (cur_track->media_info.media_type != main_media_type)
 			{
-				result_size += track->file_info.uri.len;
+				continue;
 			}
-			else
+			max_video_stream_inf += 
+				sizeof(m3u8_stream_inf_video) - 1 + 3 * VOD_INT32_LEN + MAX_CODEC_NAME_SIZE +
+				MAX_CODEC_NAME_SIZE + 1 +
+				sizeof(m3u8_stream_inf_suffix) - 1;
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: this calculation is constant please move it back to the beginning of the function

File: vod/hls/m3u8_builder.c [hunk 2]
```diff
@@ -511,48 +511,56 @@ m3u8_builder_build_master_playlist(
...
+				}
+				else
+				{
+					result_size += media_set->uri.len;
+				}
 			}
+			result_size += conf->index_file_name_prefix.len;
+			result_size += sizeof("-f-v-a") - 1 + VOD_INT32_LEN * 3;
+			result_size += sizeof(m3u8_url_suffix) - 1;
+			result_size += 2;
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: what is the purpose of this ?

File: vod/hls/m3u8_builder.c [hunk 3]
```diff
@@ -565,78 +573,89 @@ m3u8_builder_build_master_playlist(
...
-		if (cur_sequence_tracks[MEDIA_TYPE_VIDEO] != NULL)
-		{
-			track = cur_sequence_tracks[MEDIA_TYPE_VIDEO];
-			video = &track->media_info;
-			bitrate = video->bitrate;
-			if (cur_sequence_tracks[MEDIA_TYPE_AUDIO] != NULL)
+		main_media_type = cur_sequence->track_count[MEDIA_TYPE_VIDEO] != 0 ? MEDIA_TYPE_VIDEO : MEDIA_TYPE_AUDIO;
+             
+             
+		cur_sequence_tracks[MEDIA_TYPE_AUDIO] = cur_sequence->filtered_clips[0].longest_track[MEDIA_TYPE_AUDIO];
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: looking at the updated code, we no longer need the cur_sequence_tracks array, we can instead add a new variable for the audio_track and use cur_track for the video. so basically:
cur_sequence_tracks[MEDIA_TYPE_VIDEO] != NULL becomes main_media_type == MEDIA_TYPE_VIDEO
cur_sequence_tracks[MEDIA_TYPE_VIDEO] becomes cur_track
cur_sequence_tracks[MEDIA_TYPE_AUDIO] becomes audio_track

File: vod/hls/m3u8_builder.c [hunk 4]
```diff
@@ -565,78 +573,89 @@ m3u8_builder_build_master_playlist(
...
+						&video->codec_name);
+				if (cur_sequence_tracks[MEDIA_TYPE_AUDIO] != NULL)
+				{
+					*p++ = ',';
+					p = vod_copy(p, audio->codec_name.data, audio->codec_name.len);
+				}
+			}
+			else
+			{
+				cur_track = cur_sequence_tracks[MEDIA_TYPE_AUDIO];
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: you are updating the loop variable inside the loop, this a bad idea and in this case unneeded, just use cur_track as is

File: vod/hls/m3u8_builder.c [hunk 5]
```diff
@@ -511,48 +511,56 @@ m3u8_builder_build_master_playlist(
...
 	media_info_t* video;
 	media_info_t* audio = NULL;
 	uint32_t sequence_index;
 	uint32_t bitrate;
 	u_char* p;
-	size_t max_video_stream_inf;
-	size_t result_size;
+	size_t          max_video_stream_inf = 0;
+	size_t          result_size = 0;
+        uint32_t        main_media_type;
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: please change the indentation to tab, and remove the multiple spaces that prefix the variable name, the convention is to use a single space. you can search your updated file for '    ' (4 spaces) to make sure you always use tabs

File: vod/hls/m3u8_builder.c [hunk 6]
```diff
@@ -511,48 +511,55 @@ m3u8_builder_build_master_playlist(
...
+	size_t result_size = 0;
+        uint32_t main_media_type;
+        media_track_t* cur_track;
+	media_track_t* audio_track = NULL;
+
+
 
 	// calculate the result size
-	max_video_stream_inf = 
+	// calculate the result size
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: please remove the duplicate comment and the duplicate newlines that precede it (use only one newline)

File: vod/hls/m3u8_builder.c [hunk 7]
```diff
@@ -511,48 +511,55 @@ m3u8_builder_build_master_playlist(
...
-
-		track = (cur_sequence_tracks[MEDIA_TYPE_VIDEO] != NULL ?
-			cur_sequence_tracks[MEDIA_TYPE_VIDEO] :
-			cur_sequence_tracks[MEDIA_TYPE_AUDIO]);
-		if (base_url->len != 0)
-		{
-			result_size += base_url->len + 1;
-			if (track->file_info.uri.len > 0)
+	{	    
+		if(audio_track == NULL)
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: remove this if, you should always take the audio track from the current sequence

File: vod/hls/m3u8_builder.c [hunk 8]
```diff
@@ -511,48 +511,55 @@ m3u8_builder_build_master_playlist(
...
-	media_track_t* track;
+	media_sequence_t* cur_sequence;	
 	media_info_t* video;
 	media_info_t* audio = NULL;
 	uint32_t sequence_index;
 	uint32_t bitrate;
 	u_char* p;
-	size_t max_video_stream_inf;
-	size_t result_size;
+	size_t max_video_stream_inf = 0;
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: no need to initialize max_video_stream_inf, result_size, audio_track (after removing the if that i requested below)

File: vod/hls/m3u8_builder.c [hunk 9]
```diff
@@ -511,48 +511,55 @@ m3u8_builder_build_master_playlist(
...
+	size_t result_size = 0;
+        uint32_t main_media_type;
```
Comment (erankor) [inline vod/hls/m3u8_builder.c]: these 2 lines still uses space, please change to tab

**Pull request merged**